### DISCUSSION TITLE
How to debug failed login attempts

### DISCUSSION THREAD

**Discussion by iovreche** [Web3Auth/Web3Auth#1230] [Feature Requests]
I am not sure if I 'm the only one having these struggles. I have a working implementation of web3auth/react-native with custom-auth pointing to auth0. From time to time, the login fails. But I can't know why, because the error happens at the emulator 's browser. If I go back to the app, the SDK will just signal that the user canceled the login. And the error shown in the browser will just signal something along the lines of "there is a bug"... this is not useful information

I 'd like to have some place where I can check a repository of logs thrown by web3auth, at every stage of the implementation, be it brower, app, whatever. But give me something, please. I think the dashboard could be a good one.

**Comment by iovreche** [comment]
Still need help with this.

**Comment by iovreche** [comment]
Would the `redirect_uri` of `ExtraLoginOptions` be of use for this purpose?

```ts
export interface ExtraLoginOptions extends BaseLoginOptions {
    /**
     * Your Auth0 account domain such as `'example.auth0.com'`,
     * `'example.eu.auth0.com'` or , `'example.mycompany.com'`
     * (when using [custom domains](https://auth0.com/docs/custom-domains))
     */
    domain?: string;
    /**
     * The Client ID found on your Application settings page
     */
    client_id?: string;
    /**
     * The default URL where Auth0 will redirect your browser to with
     * the authentication result. It must be whitelisted in
     * the "Allowed Callback URLs" field in your Auth0 Application's
     * settings. If not provided here, it should be provided in the other
     * methods that provide authentication.
     */
    redirect_uri?: string;
    /**
     * The value in seconds used to account for clock skew in JWT expirations.
     * Typically, this value is no more than a minute or two at maximum.
     * Defaults to 60s.
     */
    leeway?: number;
    /**
     * The field in jwt token which maps to verifier id
     */
    verifierIdField?: string;
    /**
     * Whether the verifier id field is case sensitive
     * @defaultValue true
     */
    isVerifierIdCaseSensitive?: boolean;
}
```

**Comment by YZhenY** [comment]
We're working on better error fallback reporting back to the app. expect updates soon

**Reply by iovreche** [reply]
> We're working on better error fallback reporting back to the app. expect updates soon

Do we have something yet?
